### DISCUSSION TITLE
How to record the duration of the entire experiment and store it in the data CSV?

### DISCUSSION THREAD

**Discussion by zz112duke** [jspsych/jsPsych#1437] [Q&A] [answered]
Hi, 

I want to record the duration of the entire experiment and store it in the data CSV along with all of the other data.

I'm using` jsPsych.data.checks = jsPsych.data.getInteractionData().last(1)` but it doesn't appear in my final CSV file. 

The code to save data in CSV is:
`function save_data_csv() {
    jQuery.ajax({
        type: 'post',
        cache: false,
        url: save_url,
        data: {
            data_dir: data_dir,
            file_name: file_name + '.csv', // the file type should be added
            exp_data: jsPsych.data.get().csv()
        }
    });
}

jsPsych.init({
    timeline: timeline,
    display_element: 'display_stage',
    preload_images: preload_list,
    on_finish: function () {
        save_data_csv();
    }
});`

Thanks for the help!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @zz112duke,

This is already saved in the data by default. A column called `total_time_elapsed` is saved with each trial, which records the time since the start of the experiment to the end of that trial. The final trial will be the duration of the entire experiment.

**Reply by zz112duke** [reply]
It turned out that I had too many columns so I missed it. Thank you so much!
